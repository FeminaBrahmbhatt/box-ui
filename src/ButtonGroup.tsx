import React, { FC } from "react";
import { IButtonItem } from "./@interfaces";
import classNames from "classnames";

export interface ButtonGroupProps {
  options: IButtonItem[];
  activeOption: string;
  setActiveOption: (activeOption: string) => void;
}

export const ButtonGroup: FC<ButtonGroupProps> = ({
  options,
  activeOption,
  setActiveOption,
}) => {
  return (
    <div>
      {options.map((option, index) => (
        <button
          key={option.value}
          className={classNames(
            "inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500",
            {
                "rounded-l-lg": index === 0,
                "rounded-r-lg": index === options.length - 1,
                "bg-gray-50 dark:bg-gray-700": activeOption === option.value,
                "border-r-0": index !== 0 && index !== options.length - 1
            }
          )}
          onClick={() => setActiveOption(option.value)}
        >
          {option.content}
        </button>
      ))}
    </div>
  );
};
