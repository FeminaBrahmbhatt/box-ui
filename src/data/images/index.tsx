export type CountryISO = "IN" | "BE" | "DE" | "FR" | "IT" | "NL" | "RU" | "US";

type image =  CountryISO ;

// Development images
// const devImages: Record<image, string> = {
//   logo: require("./logo.png"),
//   demoAvatar: require("./demoAvatar.png"),

//   IN: require("./in.svg"),
//   BE: require("./be.svg"),
//   DE: require("./de.svg"),
//   FR: require("./fr.svg"),
//   IT: require("./it.svg"),
//   NL: require("./nl.svg"),
//   RU: require("./ru.svg"),
//   US: require("./us.svg"),

// };

// Production images (CDN)
const prodImages: Record<image, string> = {
  IN: "https://kapowaz.github.io/circle-flags/flags/in.svg",
  BE: "https://kapowaz.github.io/circle-flags/flags/be.svg",
  DE: "https://kapowaz.github.io/circle-flags/flags/de.svg",
  FR: "https://kapowaz.github.io/circle-flags/flags/fr.svg",
  IT: "https://kapowaz.github.io/circle-flags/flags/it.svg",
  NL: "https://kapowaz.github.io/circle-flags/flags/nl.svg",
  RU: "https://kapowaz.github.io/circle-flags/flags/ru.svg",
  US: "https://kapowaz.github.io/circle-flags/flags/us.svg",
};

export const images = prodImages;
// export const images = process.env.NODE_ENV === "production" ? prodImages : devImages;
