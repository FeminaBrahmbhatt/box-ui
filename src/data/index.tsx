export * from "./colors";
export * from "./countries";
export * from "./prices";
export * from "./navItems";
export * from "./options";
