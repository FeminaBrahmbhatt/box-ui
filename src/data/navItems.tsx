import React from "react";
import { INavItem } from "../@interfaces";
import {
  FiArrowLeft,
  FiBarChart2,
  FiCheckSquare,
  FiHome,
  FiLayers,
  FiSettings,
  FiFlag,
  FiMapPin,
} from "react-icons/fi";

const Home = FiHome as React.FC<any>;
const Layers = FiLayers as React.FC<any>;
const Settings = FiSettings as React.FC<any>;
const Flag = FiFlag as React.FC<any>;
const MapPin = FiMapPin as React.FC<any>;
const CheckSquare = FiCheckSquare as React.FC<any>;
const BarChart = FiBarChart2 as React.FC<any>;
const ArrowLeft = FiArrowLeft as React.FC<any>;

export const navItemsTop: INavItem[] = [
  {
    order: 0,
    path: "home",
    label: "Home",
    icon: <Home />,
  },
  {
    order: 1,
    path: "dashboard",
    label: "Dashboard",
    icon: <BarChart />,
    subItems: [
      {
        order: 0,
        path: "analytics",
        label: "Analytics",
      },
      {
        order: 1,
        path: "apps",
        label: "Apps",
      },
      {
        order: 2,
        path: "learn",
        label: "Learn",
      },
    ],
  },
  {
    order: 2,
    path: "projects",
    label: "Projects",
    icon: <Layers />,
  },
  {
    order: 3,
    path: "tasks",
    label: "Tasks",
    icon: <CheckSquare />,
  },
  {
    order: 4,
    path: "reporting",
    label: "Reporting",
    icon: <Flag />,
  },
  {
    order: 5,
    path: "rentals",
    label: "Rentals",
    icon: <MapPin />,
    subItems: [
      {
        order: 0,
        path: "all",
        label: "All rentals",
      },
      {
        order: 1,
        path: "popular",
        label: "Popular",
      },
      {
        order: 2,
        path: "recent",
        label: "Recently added",
      },
    ],
  },
];

export const navItemsBottom: INavItem[] = [
  {
    order: 0,
    path: "settings",
    label: "Settings",
    icon: <Settings />,
  },
  {
    order: 1,
    label: "Close Navbar",
    icon: <ArrowLeft />,
    toggleSidebar: true,
  },
];
