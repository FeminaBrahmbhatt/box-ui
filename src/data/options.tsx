import React from 'react'
import { IButtonItem } from "../@interfaces";
import { FiCircle, FiList, FiGrid } from "react-icons/fi";


const Circle = FiCircle as React.FC<any>;

const List = FiList as React.FC<any>;

const Grid = FiGrid as React.FC<any>;

export const options1: IButtonItem[] = [
  { value: "Leading", content: "Leading" },
  { value: "Middle", content: "Middle" },
  { value: "Trailing", content: "Trailing" },
];

export const options2: IButtonItem[] = [
  {
    value: "first",
    content:  (
      <>
        <span><Circle size={20} className='mr-2'/></span>
        First
      </>
    ),
  },
  {
    value: "second",
    content: (
      <>
        <span><Circle size={20} className='mr-2'/></span>
        Second
      </>
    ),
  },
  {
    value: "third",
    content: (
      <>
        <span><Circle size={20} className='mr-2'/></span>
        Third
      </>
    ),
  },
];

export const options3: IButtonItem[] = [
  {value:'list', content:<List size={20}/>},
  {value:'grid', content:<Grid size={20}/>},
]
