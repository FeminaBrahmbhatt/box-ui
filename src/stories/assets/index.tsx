
export type CountryISO = "BE" | "DE" | "FR" | "IT" | "NL" | "RU" | "US";

type image = CountryISO;

// Development images
const devImages: Record<string, image> = {
  NL: require("./nl.svg")
};

// Production images (CDN)
// const prodImages: Record<string, image> = {
//   NL: "https://res.cloudinary.com/tailwindcss/image/upload/v1635279277/nl_tpy2ab.svg",
// };

export const assets = devImages;