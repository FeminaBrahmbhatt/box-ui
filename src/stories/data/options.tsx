import { IButtonItem } from "../@interfaces";
import { FiCircle, FiList, FiGrid } from "react-icons/fi";

export const options1: IButtonItem[] = [
  { value: "Leading", content: "Leading" },
  { value: "Middle", content: "Middle" },
  { value: "Trailing", content: "Trailing" },
];

export const options2: IButtonItem[] = [
  {
    value: "first",
    content: (
      <>
        <FiCircle size="10" className="mr-2 center" />
          First
      </>
    ),
  },
  {
    value: "second",
    content: (
      <>
        <FiCircle size="10" className="mr-2 center" />
          Second
      </>
    ),
  },
  {
    value: "third",
    content: (
      <>
        <FiCircle size="10" className="mr-2 center" />
          Third
      </>
    ),
  },
];

export const options3 = [
  {value:'list', content:<FiList size={20}/>},
  {value:'grid', content:<FiGrid size={20}/>},
]
