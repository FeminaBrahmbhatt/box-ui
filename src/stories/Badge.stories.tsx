import React from 'react';
import {BadgeProps} from '../@interfaces';
import { Badge } from '../Badge';
import {images} from '../data/images'
import {Meta, StoryObj} from '@storybook/react';
import { StoryLayout } from '../StoryLayout';
import { FiArrowRight, FiStar } from 'react-icons/fi';
import { IconWrapper } from '../utils/IconWrapper';

export default {
    title: 'Example/Badge',
    component: Badge,
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
} as Meta;
interface Props extends BadgeProps {
    darkMode: boolean;
}

export const Default:StoryObj<Props> = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        children:<span>Label</span>
    }
}

export const LeadingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        LeadingIcon: <IconWrapper Icon={FiStar} data-testid="leadingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

export const ImageIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "gray",
        size: "md",
        LeadingIcon: <img src={images.NL} alt="nl" className="w-4 h-4" data-testid="iconOnly" />
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Newzeland
                </Badge>
            </StoryLayout>
        )
    }
}

export const TrailingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "primary",
        size: "md",
        TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

