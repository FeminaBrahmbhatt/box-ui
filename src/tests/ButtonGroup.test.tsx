import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../stories/ButtonGroup.stories';
import { options1, options2 } from "../data/options";

const {
  LabelButtonGroup,
  IconButtonGroup,
  LabelIconButtonGroup
} = composeStories(stories);

describe('ðŸ“˜ ButtonGroup Storybook Stories', () => {

  test('renders LabelButtonGroup', () => {
    const Story = LabelButtonGroup;
    const { container } = render(Story(Story.args ?? {}));

    const labelTexts = ['Leading', 'Middle', 'Trailing'];

    labelTexts.forEach((val) => {
      expect(screen.getByText(val)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  test('renders IconButtonGroup with icons', () => {
    const Story = IconButtonGroup;
    const { container } = render(Story(Story.args ?? {}));

    expect(container.querySelector('svg')).toBeInTheDocument();  // Check for an icon (List or Grid)
    expect(container).toMatchSnapshot();
  });

  test('renders LabelIconButtonGroup', () => {
    const Story = LabelIconButtonGroup;
    const { container } = render(Story(Story.args ?? {}));

    const labelTexts = ['First', 'Second', 'Third'];

    labelTexts.forEach((text) => {
      expect(screen.getByText(text)).toBeInTheDocument();
    });
    expect(container.querySelector('svg')).toBeInTheDocument(); // Check for icons

    expect(container).toMatchSnapshot();
  });

});
