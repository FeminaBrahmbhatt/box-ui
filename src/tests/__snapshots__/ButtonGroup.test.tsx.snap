// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“˜ ButtonGroup Storybook Stories renders IconButtonGroup with icons 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-l-lg bg-gray-50 dark:bg-gray-700"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="8"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="8"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="8"
              x2="21"
              y1="18"
              y2="18"
            />
            <line
              x1="3"
              x2="3.01"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="3.01"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="3.01"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-r-lg"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="7"
              width="7"
              x="3"
              y="3"
            />
            <rect
              height="7"
              width="7"
              x="14"
              y="3"
            />
            <rect
              height="7"
              width="7"
              x="14"
              y="14"
            />
            <rect
              height="7"
              width="7"
              x="3"
              y="14"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ ButtonGroup Storybook Stories renders LabelButtonGroup 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-l-lg"
        >
          Leading
        </button>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 bg-gray-50 dark:bg-gray-700 border-r-0"
        >
          Middle
        </button>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-r-lg"
        >
          Trailing
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ ButtonGroup Storybook Stories renders LabelIconButtonGroup 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-l-lg"
        >
          <span>
            <svg
              class="mr-2"
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
          </span>
          First
        </button>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 bg-gray-50 dark:bg-gray-700 border-r-0"
        >
          <span>
            <svg
              class="mr-2"
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
          </span>
          Second
        </button>
        <button
          class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-center text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-r-lg"
        >
          <span>
            <svg
              class="mr-2"
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
          </span>
          Third
        </button>
      </div>
    </div>
  </div>
</div>
`;
